payment/payment.go

```diff
@@ -146,22 +146,21 @@ func FromScript(
 	witnessScriptHash := make([]byte, 0)
 	_script := make([]byte, 0)
 	witnessScript := make([]byte, 0)
-	switch script[0] {
-	// p2wpkh / p2wsh
-	case txscript.OP_0:
-		// in case the script is a p2wpkh, we need to set also the legacy script
+	switch address.GetScriptType(script) {
+	case address.P2WpkhScript:
 		if len(script[2:]) == 20 {
 			scriptHash = append(scriptHash, script[2:]...)
 			_script = buildScript(scriptHash, "p2pkh")
 		}
```

Comment:
This statement checks that the script is of type p2wpkh actually.
With the new changes, this if can be deleted since we are sure the script is a p2wpkh in this case section.
```suggestion
		scriptHash = append(scriptHash, script[2:]...)
		_script = buildScript(scriptHash, "p2pkh")
```

Reply:
Done